src/main/java/net/fosforito/partido/api/UsersApi.java

```diff
@@ -70,26 +66,12 @@ public User getCurrentUser() {
    */
   @PostMapping(value = {"/users"}, produces = MediaType.APPLICATION_JSON, consumes = MediaType.APPLICATION_JSON)
   public ResponseEntity<User> createUser(@RequestBody UserDTO userDTO) {
-    User user = new User();
-    if (userRepository.findByEmail(userDTO.getEmail()) != null) {
-      // Email already registered! Return error status
-      // with empty user object because client handling is problematic else
-      return new ResponseEntity<>(user, HttpStatus.PRECONDITION_FAILED);
+    Pair<String, User> validationOfCreatedUser = userService.createUser(userDTO);
+    if (validationOfCreatedUser.getFirst() == null) {
+      return new ResponseEntity<>(validationOfCreatedUser.getSecond(), HttpStatus.OK);
+    } else {
+      return ResponseEntity.status(HttpStatus.CONFLICT).build();
```

Comment:
The conflict status code is definitely better than precondition failed. I'll create an issue because this affects the client app where it must be changed as well. 👌
Edit: new issue partidodev/partido-client#22